indigo_app/views/works.py

```diff
@@ -1346,16 +1345,69 @@ def get_context_data(self, **kwargs):
             repeal_made = self.request.POST.get('repeal_made')
             if repeal_made:
                 repealed_work = Work.objects.filter(pk=repeal_made).first()
-                if repealed_work:
+                if repealed_work and repealed_work.repealed_by != self.work and not any(
+                        [True for i in initial if i["repealed_work"] == repealed_work]):
                     initial.append({
-                         'repealed_work': repealed_work,
-                         'repealed_date': repealed_work.repealed_date,
+                        'repealed_work': repealed_work,
+                        'repealed_date': repealed_work.repealed_date or self.work.commencement_date,
                     })
 
         context["form"] = {
-           'repeals_made_formset': RepealMadeBaseFormSet(prefix='repeals_made', initial=initial, form_kwargs={"work": self.work}),
+           'repeals_made_formset': RepealMadeBaseFormSet(
+               prefix='repeals_made',
+               initial=initial,
+               form_kwargs={"work": self.work}
+           ),
         }
         context["work"] = self.work
         return context
 
 
+class WorkFormAmendmentsView(WorkViewBase, TemplateView):
+    template_name = 'indigo_api/_work_form_amendments_form.html'
+
+    def post(self, request, *args, **kwargs):
+        return self.get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        formset = AmendmentsBaseFormSet(self.request.POST, prefix="amendments", form_kwargs={"work": self.work})
+        initial = []
+        if formset.is_valid():
+            for form in formset:
+                delete = form.cleaned_data.get('DELETE')
+                if delete:
+                    if not Amendment.objects.filter(
+                            amended_work=self.work,
+                            amending_work=form.cleaned_data["amending_work"]
+                    ).exists():
+                        continue
+                initial.append({
+                    "amended_work": self.work,
+                    "amending_work": form.cleaned_data["amending_work"],
+                    "date": form.cleaned_data["date"],
+                    "DELETE": form.cleaned_data["DELETE"],
+                })
+            amendment = self.request.POST.get("amendment")
+            if amendment:
+                amending_work = Work.objects.filter(pk=amendment).first()
+                if amending_work:
+                    # add amendment only if it does not exist and is not in the initial data
+                    if not Amendment.objects.filter(amended_work=self.work, amending_work=amending_work).exists() and not any(
+                            [True for i in initial if i["amending_work"] == amending_work]):
```

Comment:
```suggestion
                    in_initial_dates = [i["date"] for i in initial if i["amending_work"] == amending_work]
                    existing_amendments = Amendment.objects.filter(amended_work=self.work, amending_work=amending_work, date__in=in_initial_dates).all()
                    if not existing_amendments:
```
This is a toughie, damn! 😂 
I'm not sure if this does the right thing, but what we want is to be able to add multiple amendments by the same work, as long as they're at different dates.
So the only thing we don't want to do is to add another amendment by the same work at the same date, because then we'll get issues on an attempted save.
(Another thought: maybe if this is not the first amendment by this amending work, we actually don't give a default date?)

Reply:
oh, another thing we might do here is to only look in `initial` for another amendment by the same amending work _that doesn't yet have a date associated_ — and if there is one, don't add another one?

Reply:
An amendment happens through an associated object. Why can we store the pk of that object in the related form? When the form is saved, we know which Amendment objects must be update or created based on this pk.

Reply:
haha yes that sounds much better 😅